src/main/scala/com/amazon/deequ/analyzers/runners/AnalysisRunner.scala

```diff
@@ -404,6 +404,9 @@ object AnalysisRunner {
     passedAnalyzers.foreach { analyzer =>
       stateLoaders.foreach { stateLoader =>
         analyzer.aggregateStateTo(aggregatedStates, stateLoader, aggregatedStates)
+        if(saveStatesWith.isDefined) {
```

Comment:
I dont think this is the correct place to persist the aggregates, as these are only the initial aggregates before the actual delta of the data is processed.

Reply:
Probably I'm probably not following properly the code  (sorry, I'm not familiarised with Scala).

After this point I can't see any modification on the aggregated states for the "scanningAnalyzers", as they only read this states but not modify them.

For "groupingAnalyzers" they are actually modified but they also override them on the "saveStatesWith" so the final result is the expected one. 

Please let me know what would be the proper place to locate it and I will move.

Reply:
Hi Roman,

In general your code would be correct. The problem is just that the states for the grouping analyzers can become quite large and we need to be careful when persisting them. We have an optimization where we only persist large states once if they can be reused by multiple grouping analyzers.